lib/db/DB.ts

```diff
@@ -58,8 +58,22 @@ class DB {
       const { dialectName } = this.sequelize.connectionManager;
       this.logger.info(`connected to database. host:${host} port:${port} database:${database} dialect:${dialectName}`);
     } catch (err) {
-      this.logger.error('unable to connect to the database', err);
-      throw err;
+      if (err instanceof Sequelize.InvalidConnectionError) {
+        try {
+          const initSequelize = new Sequelize('', this.sequelize.options.username, this.sequelize.options.password, this.sequelize.options);
+          await initSequelize.authenticate();
+          await initSequelize.query(`CREATE DATABASE IF NOT EXISTS ${this.sequelize.options.database};`);
+          await initSequelize.query(`GRANT ALL PRIVILEGES ON ${this.sequelize.options.database}.* TO '${this.sequelize.options.username}'@'%'`);
+          await initSequelize.close();
```

Comment:
Any reason for not continuing with simply one connection, by defining a database context?

Reply:
I wasn't able to find a way how to do this.

Reply:
`USE [database]` doesn't work?

Reply:
Good call, this works for me:

```javascript

        this.sequelize = new Sequelize('', this.sequelize.options.username, this.sequelize.options.password, this.sequelize.options);
        await this.sequelize.authenticate();
        await this.sequelize.query(`CREATE DATABASE IF NOT EXISTS ${this.sequelize.options.database};`);
        this.sequelize.query(`USE ${this.sequelize.options.database}`);
```